### PR TITLE
Update to Net5.0 and use Nuke build

### PR DESCRIPTION
**Pull request by slewis74**
Easiest to review by commit.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+# Controls when the action will run. 
+on: 
+  # Triggers the workflow on pull request events and merges/pushes to main
+  push:
+    branches:
+      - main
+      - release/*
+  pull_request:
+    types: [opened, synchronize, reopened]
```
Comment (rain-on) [inline .github/workflows/main.yml:11]: think this is the default - maybe not required?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-18.04
```
Comment (rain-on) [inline .github/workflows/main.yml:18]: do we want to go with 20.04, 18.04 is kinda old these days,.

File: .nuke/parameters.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "$schema": "./build.schema.json",
+  "Solution": "source/AzureDevOpsIssueTracker.sln"
```
Comment (rain-on) [inline .nuke/parameters.json:3]: nit: does this need a ./ prefix as per line-2?

File: build.ps1 [hunk 4]
```diff
@@ -1,189 +1,69 @@
...
-Write-Host "Preparing to run build script..."
+###########################################################################
+# CONFIGURATION
+###########################################################################
 
-if(!$PSScriptRoot){
-    $PSScriptRoot = Split-Path $MyInvocation.MyCommand.Path -Parent
-}
+$BuildProjectFile = "$PSScriptRoot\build\_build.csproj"
+$TempDirectory = "$PSScriptRoot\\.nuke\temp"
```
Comment (rain-on) [inline build.ps1:17]: double "\\", but not on line 16

File: build.sh [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+set -eo pipefail
+SCRIPT_DIR=$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)
+
+###########################################################################
+# CONFIGURATION
+###########################################################################
+
+BUILD_PROJECT_FILE="$SCRIPT_DIR/build/_build.csproj"
+TEMP_DIRECTORY="$SCRIPT_DIR//.nuke/temp"
```
Comment (rain-on) [inline build.sh:13]: double "//", and line 15

File: build.sh [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+export DOTNET_CLI_TELEMETRY_OPTOUT=1
+export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
+export DOTNET_MULTILEVEL_LOOKUP=0
+
+###########################################################################
+# EXECUTION
+###########################################################################
+
+function FirstJsonValue {
+    perl -nle 'print $1 if m{"'"$1"'": "([^"]+)",?}' <<< "${@:2}"
```
Comment (rain-on) [inline build.sh:28]: definitely worth updating this to use jq

File: .github/workflows/main.yml (hunk 1 above)
Comment (hnrkndrssn) [reply .github/workflows/main.yml:11]: yes, those are the default values so that _could_ be dropped

File: .github/workflows/main.yml (hunk 2 above)
Comment (hnrkndrssn) [reply .github/workflows/main.yml:18]: `ubuntu-latest` would be nice (this is currently `20.04`)

**Review by hnrkndrssn — APPROVED** [review]
Apart from a few :octocat: workflow file changes and it being mostly copy 🍝 from already established code, this looks fine

File: .nuke/parameters.json (hunk 3 above)
Comment (slewis74) [reply .nuke/parameters.json:3]: I don't think so. None of the others have it, and Nuke is .NET based, so pretty sure won't be picky about the leading `./` on that path. `$schema` is different, which does seem odd when they're right next to each other.

File: build.ps1 (hunk 4 above)
Comment (slewis74) [reply build.ps1]: Fixed

File: build.sh (hunk 5 above)
Comment (slewis74) [reply build.sh]: Fixed

File: .github/workflows/main.yml (hunk 2 above)
Comment (slewis74) [reply .github/workflows/main.yml:18]: Sticking with `18.04` for the moment, a library OctoVersion uses for Git access is throwing an exception on `20.04`

File: .github/workflows/main.yml (hunk 1 above)
Comment (slewis74) [reply .github/workflows/main.yml]: Done

File: .github/workflows/main.yml (hunk 2 above)
Comment (slewis74) [reply .github/workflows/main.yml]: Fixed, we're on `20.04`

File: build.sh (hunk 6 above)
Comment (rain-on) [reply build.sh:28]: but probably not on this commit :)

**Pull request merged**